src/Features/CSharp/Portable/CodeFixes/ConditionalExpressionInStringInterpolation/CSharpAddParenthesesAroundConditionalExpressionInInterpolatedStringCodeFixProvider.cs

```diff
@@ -64,18 +64,52 @@ private static async Task<Document> GetChangedDocumentAsync(Document document, i
             var openParenthesisPosition = conditionalExpressionSyntaxStartPosition;
             var textWithOpenParenthesis = text.Replace(openParenthesisPosition, 0, "(");
             var documentWithOpenParenthesis = document.WithText(textWithOpenParenthesis);
+
             var syntaxRoot = await documentWithOpenParenthesis.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
             var nodeAtInsertPosition = syntaxRoot.FindNode(new TextSpan(openParenthesisPosition, 0));
-            if (nodeAtInsertPosition is ParenthesizedExpressionSyntax parenthesizedExpression &&
-                parenthesizedExpression.CloseParenToken.IsMissing)
+
+            if (nodeAtInsertPosition is not ParenthesizedExpressionSyntax parenthesizedExpression ||
+                !parenthesizedExpression.CloseParenToken.IsMissing)
+            {
+                return documentWithOpenParenthesis;
+            }
+
+            return await InsertCloseParenthesisAsync(
+                documentWithOpenParenthesis, parenthesizedExpression, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<Document> InsertCloseParenthesisAsync(
+            Document document,
+            ParenthesizedExpressionSyntax parenthesizedExpression,
+            CancellationToken cancellationToken)
+        {
+            var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            if (parenthesizedExpression.Expression is ConditionalExpressionSyntax conditional &&
+                parenthesizedExpression.GetAncestor<InterpolatedStringExpressionSyntax>()?.StringStartToken.Kind() == SyntaxKind.InterpolatedStringStartToken)
             {
-                var newCloseParen = SyntaxFactory.Token(SyntaxKind.CloseParenToken).WithTriviaFrom(parenthesizedExpression.CloseParenToken);
-                var parenthesizedExpressionWithClosingParen = parenthesizedExpression.WithCloseParenToken(newCloseParen);
-                syntaxRoot = syntaxRoot.ReplaceNode(parenthesizedExpression, parenthesizedExpressionWithClosingParen);
-                return documentWithOpenParenthesis.WithSyntaxRoot(syntaxRoot);
+                // If they have something like:
+                //
+                // PreviousLineOfCode();
+                // var s3 = $""Text1 { true ? ""Text2""[|:|]
```

Comment:
I didn't follow the example. We need some parenthesized expression containing a conditional to get here

Reply:
updated the comment to explain this.

Reply:
Thanks!